crates/node/builder/src/launch/engine.rs

```diff
@@ -202,12 +202,28 @@ where
         let pruner_events = pruner.events();
         info!(target: "reth::cli", prune_config=?ctx.prune_config().unwrap_or_default(), "Pruner initialized");
 
-        // TODO: implement methods which convert this value into an actual function
-        if let Some(ref hook_type) = ctx.node_config().debug.invalid_block_hook {
-            warn!(target: "reth::cli", ?hook_type, "Invalid block hooks are not implemented yet! The `debug.invalid-block-hook` flag will do nothing for now.");
-        }
-
-        let invalid_block_hook = Box::new(NoopInvalidBlockHook);
+        let invalid_block_hook: Box<dyn InvalidBlockHook> = if let Some(ref hook) =
+            ctx.node_config().debug.invalid_block_hook
+        {
+            warn!(target: "reth::cli", ?hook, "Invalid block hooks are not implemented yet! The `debug.invalid-block-hook` flag will do nothing for now.");
+            Box::new(InvalidBlockHookChain(
+                hook.to_selection()
+                    .into_iter()
```

Comment:
isn't this just iter().copied() ?

Reply:
yep, added a `Deref` impl to the selection struct and replace with `iter().copied()` here